internal/api/api.go

```diff
@@ -57,11 +59,26 @@ func acceptedHosts() []string {
 // Listener returns the default listener if OCTANT_LISTENER_ADDR is not set.
 func Listener() (net.Listener, error) {
 	listenerAddr := getListenerAddr()
-	conn, err := net.DialTimeout("tcp", listenerAddr, time.Millisecond*500)
+	host, port, err := net.SplitHostPort(listenerAddr)
 	if err != nil {
-		return net.Listen("tcp", listenerAddr)
+		panic(fmt.Sprintf("Unable to parse OCTANT_LISTENER_ADDR: %s", listenerAddr))
+	}
+	portNumber, err := strconv.Atoi(port)
+	if err != nil {
+		panic(fmt.Sprintf("Unable to convert port: %s", port))
+	}
+	for i := 0; ; i++ {
+		port := portNumber + i
+		listenerAddr = net.JoinHostPort(host, fmt.Sprintf("%d", port))
```

Comment:
If `OCTANT_LISTENER_ADDR` or `--listener-addr` is explicitly provided, we want to fail. We don't want to randomize ports when explicitly given a value.